src/uint/modular/constant_mod.rs

```diff
@@ -143,3 +150,40 @@ impl<MOD: ResidueParams<LIMBS>, const LIMBS: usize> Retrieve for Residue<MOD, LI
         self.retrieve()
     }
 }
+
+#[cfg(feature = "serde")]
+impl<'de, MOD, const LIMBS: usize> Deserialize<'de> for Residue<MOD, LIMBS>
+where
+    MOD: ResidueParams<LIMBS>,
+    Uint<LIMBS>: Encoding,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Uint::<LIMBS>::deserialize(deserializer).and_then(|montgomery_form| {
+            if Uint::ct_lt(&montgomery_form, &MOD::MODULUS).into() {
+                Ok(Self {
+                    montgomery_form,
+                    phantom: PhantomData,
+                })
+            } else {
+                Err(D::Error::custom("montgomery form must be reduced"))
+            }
+        })
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de, MOD, const LIMBS: usize> Serialize for Residue<MOD, LIMBS>
+where
+    MOD: ResidueParams<LIMBS>,
+    Uint<LIMBS>: Encoding,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.montgomery_form.serialize(serializer)
```

Comment:
Wondering if it would make sense to convert out of Montgomery form prior to serialization. I think it's atypical to serialize Montgomery form directly to the wire.

cc @fjarri

Reply:
Converting would induce a performance penalty in my application.

Reply:
Yes, converting out would entail a small performance penalty

Reply:
I think it is more straightforward to serialize in Montgomery form. Those who want conversion can convert manually, but if we convert automatically there will be no way to provide the no-conversion functionality.

Reply:
OK, sounds good